### PR TITLE
Send Log statements to log Topic for Java Functions

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation

Currently logTopic directive only works for Python functions. This pr brings that functionality to Java Functions as well.

### Modifications

Describe the modifications you've done.

### Result

After your change, what will change.

### CODE REVIEW

**Comment by srkukarni** [comment]
@sijie @merlimat

**Comment by srkukarni** [comment]
retest this please

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java [hunk 1]
```diff
@@ -262,10 +268,12 @@ public void run() {
                 }
                 long processAt = System.currentTimeMillis();
                 stats.incrementProcessed(processAt);
+                addLogTopicHandler();
```
Comment (sijie) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java]: this seems to be done at per message?

Can we configure the log handler at the beginning of the instance and remove it at the end?

Comment (srkukarni) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java]: We need to send only the log messages inside the function to the log topic. Other logging messages inside the java instance or runtime should not be sent. Thats why we need to do this on a per message basis.

Comment (sijie) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java]: okay... I am not sure if there is any performance penalty here. but okay to me for now.

**Pull request merged**